android/src/main/java/com/twiliorn/library/FrameCaptureCameraCapturer.java

```diff
@@ -0,0 +1,80 @@
+package com.twiliorn.library;/*
+ * Copyright © 2024. Streem, Inc. All rights reserved.
+ */
+
+import android.content.Context;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.facebook.react.bridge.ReactContext;
+import com.twilio.video.CameraCapturer;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import tvi.webrtc.CapturerObserver;
+import tvi.webrtc.SurfaceTextureHelper;
+import tvi.webrtc.VideoFrame;
+
+public class FrameCaptureCameraCapturer extends CameraCapturer {
```

Comment:
Custom CameraCapturer that allows capturing a frame.  When initialized, it creates a CapturerObserver that just serves as a pass-through to the upstream one (managed by the LocalVideoTrack that created this capturer) -- with the addition that it can intercept and save a frame when requested.

I copied the code from FrameCaptureVideoSink for intercepting a frame when requested, doing the needed conversions, and saving it to disk.